docs/adr/00XX-bulk-data-loading.md

```diff
@@ -0,0 +1,95 @@
+# XX. Bulk data loading
+
+Date: 2019-12-19
+
+## Context
+
+Content Publisher regularly has to present users with long lists of data that
+is loaded from the Publishing API, examples of these are organisations,
+contacts and the topic taxonomy. Content Publisher typically needs to
+download all the records of a particular datatype, which can involve
+multiple HTTP requests. When this is done during a web request the application
+is at an increased risk of returning a slow response or even timing out.
+The risks of this have been partially mitigated by caching.
+
+These situations where Content Publisher needs all the data records for a
+particular dataset are colloquially referred to as bulk data. This is
+distinguished from other API look ups where we may need to access a
+single resource (such as looking up the topic tags) that have less risks of
+performance penalties.
+
+This particular problem has been faced and solved for GOV.UK before. Whitehall
+takes an [approach](https://github.com/alphagov/whitehall/pull/3298) for the
+topic taxonomy where the data is loaded at runtime from
+[Redis](https://redis.io/). A period background job then refreshes the
+data in Redis. This avoids the need to look up the data during a web request.
+
+## Decision
+
+In Content Publisher we have chosen to implement a similar pattern to Whitehall.
+Data is read from Redis at runtime and a background process runs a periodic
+job to re-populate the Redis data.
+
+This approach makes use of the [Rails Redis cache store][redis-cache-store]
+which is wrapped within an object, [BulkData::Cache][bulk-data-cache], for
+writing and accessing data. We have introduced a concept of
+[repository][government-repository] classes that can be used to work with the
+loading and modelling of data that is stored in the cache.
+
+On a 15 minute interval a [background job][populate-bulk-data-job] will run to
+try re-populate the data. This job will be initially triggered each time the
```

Comment:
Could you try to reword these two sentences as I think they're a little unclear.
For example, what does "job" refer to here? I don't think it's the same as the "background job"?

Reply:
Thanks I've rephrased this.